src/main/java/graphql/nadel/engine/HydrationInputResolver.java

```diff
@@ -452,9 +452,9 @@ private ObjectExecutionResultNode findMatchingResolvedNode(ExecutionContext exec
         String inputNodeId = (String) inputNode.getCompletedValue();
         for (ExecutionResultNode resolvedNode : resolvedNodes) {
             LeafExecutionResultNode idNode = getFieldByResultKey((ObjectExecutionResultNode) resolvedNode, objectIdentifier);
-            assertNotNull(idNode, "no value found for object identifier: " + objectIdentifier);
+            assertNotNull(idNode, () -> String.format("no value found for object identifier: " + objectIdentifier));
```

Comment:
bad error here

either all string concat or all .format but not a mix

Reply:
changed